math/src/polynom/univariate.rs

```diff
@@ -0,0 +1,263 @@
+use alloc::vec::Vec;
+
+use utils::{ByteReader, ByteWriter, Deserializable, DeserializationError, Serializable};
+
+use crate::{batch_inversion, polynom, FieldElement};
+
+// COMPRESSED UNIVARIATE POLYNOMIAL
+// ================================================================================================
+
+/// The coefficients of a univariate polynomial of degree n with the linear term coefficient
+/// omitted.
+///
+/// This compressed representation is useful during the sum-check protocol as the full uncompressed
+/// representation can be recovered from the compressed one and the current sum-check round claim.
+#[derive(Clone, Debug)]
+pub struct CompressedUnivariatePolyCoef<E: FieldElement>(Vec<E>);
+
+impl<E: FieldElement> CompressedUnivariatePolyCoef<E> {
+    /// Evaluates a polynomial at a challenge point using a round claim.
+    ///
+    /// The round claim is used to recover the coefficient of the linear term using the relation
+    /// 2 * c0 + c1 + ... c_{n - 1} = claim. Using the complete list of coefficients, the polynomial
+    /// is then evaluated using Horner's method.
+    pub fn evaluate_using_claim(&self, claim: &E, challenge: &E) -> E {
```

Comment:
nit: should we rename `challenge` to `x` as we usually use `x` for variables at which we evaluate polynomials?